stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemUtils.java

```diff
@@ -49,41 +47,31 @@
     private static final String UNKNOWN_MESSAGE = "Unknown Error";
 
     /**
-     * Returns error object for input reason.
-     * Error type is generic ballerina error type. This utility to construct error object from message.
+     * Returns error object  with message. Error type is generic ballerina error type. This utility to construct
+     * error object from message.
      *
-     * @param error Reason for creating the error object. If the reason is null, "UNKNOWN" sets by
-     *              default.
+     * @param typeId The string type ID of the particular error object. If the typeId is  null, "UNKNOWN" sets by
```

Comment:
It's is required that the typeId to be same name as the defined type for distinct types.
For example `type A distinct error`, here typeId of `A` is `A`. 
Please refer the spec, in the spec the value we call typeId is referred to as `local id`.
[https://ballerina.io/ballerina-spec/spec.html#section_5.1.3](https://ballerina.io/ballerina-spec/spec.html#section_5.1.3)

Reply:
The issue is we don't have a way to enforce this when we are creating error values programmatically in Java.

Reply:
I think that param doc is misleading. I removed it. The original author must have meant to set the error message as "UNKNOWN" if the error reason(previously we had reason for errors) or the throwable is null. I fixed the logic to check only the throwable as the typeId cannot be null and throughout the module, a null has NOT been passed as a parameter to the typeId.